File: c-programs/SumPrimeDigits.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>

int main()
{

  int digits;
  int total = 0;
  scanf("%d", &digits);
  while (digits != 0)
  {

    for (int i = 1; i <= digits; i++)
    {
      int lastDigit = digits % 10;
      int totalFactors = 0;
      for (int j = 1; j <= lastDigit; j++)
      {
        if (lastDigit % j == 0 && lastDigit != 1)
        {
          totalFactors = totalFactors + 1;
        }
      }
      if (totalFactors == 2)
      {
        total = total + lastDigit;
      }
      digits = digits / 10;
    }
  }
  printf("%d", total);

  return 0;
}

File: python-programs/SumPrimeDigits.py
digits = int(input())
total = 0
while (digits != 0):
  for j in range(1, digits):
    lastDigit = digits % 10
    totalFactors = 0
    for i in range(1, lastDigit + 1):
      if lastDigit % i == 0 and lastDigit != 1:
        totalFactors = totalFactors + 1
    if totalFactors == 2:
        total = total + lastDigit    
      
    digits = digits // 10

print(total)

